sw.js

```diff
@@ -0,0 +1,61 @@
+---
+layout: null
+---
+'use strict';
+
+const version = 'v1';
+const staticCacheName = "version-{{site.time | date: '%Y%m%d%H%M%S'}}";
+
+addEventListener('install', installEvent => {
+     console.log('The service worker is installing...');
+     skipWaiting();
+     installEvent.waitUntil(
+       caches.open(staticCacheName)
+       .then( staticCache => {
+         return staticCache.addAll([
+           '/assets/css/style.css',
+           '/assets/minima-social-icons.svg',
+           '/about/index.html',
+           '/404.html',
+           'offline.html'
+         ]);
+       })
+     );
+   });
+
+addEventListener('activate', activateEvent => {
+     console.log('The service worker is activated. Deleting old caches...');
+     activateEvent.waitUntil(
+       caches.keys()
+       .then( cacheNames => {
+         return Promise.all(
+           cacheNames.map( cacheName => {
+             if (cacheName != staticCacheName) {
+               return caches.delete(cacheName);
+             }
+           })
+       );
+     })
+     .then( () => {
+       return clients.claim();
+     })
+   );
+});
+
+
+addEventListener('fetch', fetchEvent => {
+     console.log('The service worker is listening.');
+     const request = fetchEvent.request;
+     fetchEvent.respondWith(
+       caches.match(staticCacheName)
```

Comment:
This wasn't retrieving assets from the cache when offline, because it needs to be changed to `caches.match(request)`.